bundles/EcommerceFrameworkBundle/IndexService/Worker/ElasticSearch/AbstractElasticSearch.php

```diff
@@ -836,4 +840,126 @@ protected function createEsIndex(string $indexName)
         $this->settingsHash = md5(json_encode($this->tenantConfig->getIndexSettings()));
         $this->updateVersionFile();
     }
+
+    /**
+     * Get the next index version, e.g. if currently 13, then 14 will be returned.
+     * @return int
+     */
+    private function getNextIndexVersion() : int {
```

Comment:
should we make all private functions protected in order to give possibility to overwrite if necessary?

Reply:
and should we use this method in line [538](https://github.com/pimcore/pimcore/pull/5744/files#diff-d02a59391d96d7d11a3a41cfdf9270deL536) too?

Reply:
To stay consistent I declared all private methods as protected.
I also replaced ``$this->indexVersion++`` with ``getNextIndexVersion()``.